docs/src/Callout.doc.js

```diff
@@ -465,6 +465,125 @@ function Example(props) {
       `}
       />
     </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Custom navigation"
+      description={`
+These examples illustrate a custom navigation implementation to externally control the link functionality within Callout.
+
+If passed to Provider's \`onNavigation\` prop, \`useCustomOnNavigationProvider\`, a high-order function, is passed down to Callout where it's executed. Then, \`onNavigation\` returns a function that gets called during the \`onClick\` event handler.
+
+The \`useCustomOnNavigationProvider\` function can contain complex logic, including [React hooks](https://reactjs.org/docs/hooks-reference.html), to perform side effects. It takes named arguments: \`href\`, \`onClick\` and \`target\`.
+
+In the examples below, \`useCustomOnNavigationProvider\` executes the following actions:
+- Disable the default link behavior
+- Show an alert message
+- Open a different URL in a new window
+
+Finally, \`useCustomOnNavigationLink\` gets passed to ActivationCard using the \`customOnNavigation\` prop. It has the same structure as \`useCustomOnNavigationProvider\`. Callout's \`customOnNavigation\` prop also takes "disabled" to disable the Provider's \`onNavigation\` prop logic and restore the default link behaviour.
+
+The returned \`onNavigationClick\` function inside both hook functions uses the event access to [preventDefault()](https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault). It could also be used to [stopPropagation()](https://developer.mozilla.org/en-US/docs/Web/API/Event/stopPropagation).
```

Comment:
I see all the examples here use the same logic for examples. This is fine but is there a way we could share this text a bit to make sure it stays up to date without needing a manual update to every place it's used?

Reply:
DONE